File: RainMusicAndroid/app/src/main/java/com/rain/music/data/model/Song.kt
package com.rain.music.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "songs")
data class Song(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val title: String,
    val artist: String = "未知艺术家",
    val album: String = "未知专辑",
    val duration: Long = 0,         // 毫秒
    val fileUri: String,            // 文件路径
    val albumArtUri: String? = null, // 封面 URI
    val hasLyrics: Boolean = false,
    val lyricsFilePath: String? = null,
    val dateAdded: Long = System.currentTimeMillis(),
    val sourceType: String = "imported" // "library" 或 "imported"
) {
    val formattedDuration: String
        get() {
            val minutes = (duration / 1000) / 60
            val seconds = (duration / 1000) % 60
            return String.format("%d:%02d", minutes, seconds)
        }
}

File: RainMusic/Models/Song.swift
import Foundation
import SwiftData
import SwiftUI
import UIKit

@Model
final class Song {
    @Attribute(.unique) var id: UUID
    var title: String
    var artist: String
    var album: String
    var duration: Double
    var fileURL: URL
    var albumArtData: Data?
    var hasLyrics: Bool
    var lyricsFileURL: URL?
    var dateAdded: Date
    var sourceType: String // "library" 或 "imported"
    var mediaLibraryPersistentID: String?

    init(
        id: UUID = UUID(),
        title: String,
        artist: String = "未知艺术家",
        album: String = "未知专辑",
        duration: Double = 0,
        fileURL: URL,
        albumArtData: Data? = nil,
        hasLyrics: Bool = false,
        lyricsFileURL: URL? = nil,
        dateAdded: Date = Date(),
        sourceType: String = "imported",
        mediaLibraryPersistentID: String? = nil
    ) {
        self.id = id
        self.title = title
        self.artist = artist
        self.album = album
        self.duration = duration
        self.fileURL = fileURL
        self.albumArtData = albumArtData
        self.hasLyrics = hasLyrics
        self.lyricsFileURL = lyricsFileURL
        self.dateAdded = dateAdded
        self.sourceType = sourceType
        self.mediaLibraryPersistentID = mediaLibraryPersistentID
    }

    var albumArtImage: Image? {
        guard let data = albumArtData, let uiImage = UIImage(data: data) else {
            return nil
        }
        return Image(uiImage: uiImage)
    }

    var formattedDuration: String {
        let minutes = Int(duration) / 60
        let seconds = Int(duration) % 60
        return String(format: "%d:%02d", minutes, seconds)
    }
}
